src/nullable_allocator.zig

```diff
@@ -0,0 +1,31 @@
+const std = @import("std");
+
+/// A nullable allocator the same size as `std.mem.Allocator`.
+pub const NullableAllocator = struct {
+    ptr: *anyopaque = undefined,
+    // Utilize the null pointer optimization on the vtable instead of
+    // the regular ptr because some allocator implementations might tag their
+    // `ptr` property.
+    vtable: ?*const std.mem.Allocator.VTable = null,
+
+    pub inline fn new(a: std.mem.Allocator) @This() {
```

Comment:
This should be `init` instead of `new`. In Zig, the convention is for `init` to be the initializer

```suggestion
    pub inline fn init(a: std.mem.Allocator) @This() {
```